### ISSUE TITLE
Windows bots pub upgrade error - works fine locally

### ISSUE THREAD

**Issue by jakemac53** [dart-lang/package-bots#5]
see http://chromegw.corp.google.com/i/client.dart.packages/builders/packages-windows-paper-elements/builds/307/steps/annotated_steps/logs/stdio

**Comment by jakemac53** [comment]
looks like both pub upgrade and pub build are having issues finding some files

**Comment by jakemac53** [comment]
cc @ricowind

**Comment by ricowind** [comment]
cc @sigmundch : this seems like a pub problem right?

**Comment by sigmundch** [comment]
Not sure - it could be. It could also be a cache corruption issue in our cleanup in the bots. For example, recall in the past when we had a recursive delete command that was deleting inside the symlinked package folders. I thought we fixed that, but it could be something similar?

**Comment by ricowind** [comment]
So I guess we have two choices:
  -- live with the failures that our customers are also seeing
  -- run with a clean cache

I vote for the first since otherwise we will never take the time to debug
this.

The next time we see this we should debug it on the bot!

Cheers,
Rico

**Comment by sigmundch** [comment]
Yeah - agree. I think some bots right now have the failures, so we should look into it (or manually fix the cache once today, and look into it again on a fresh failure next time it fails)

**Comment by jakemac53** [comment]
It seems to happen consistently for the paper_elements package

**Comment by ricowind** [comment]
Siggi has access to log into the bot if you want to take a closer look at
the contents of the pub cache

Cheers,
Rico

**Comment by sigmundch** [comment]
Ok, we have one theory, it seems that we still use the `rmtree` in one place [here](https://github.com/dart-lang/package-bots/blob/master/annotated_steps.py#L114) where we clean up the tmp folder.

**Comment by ricowind** [comment]
That is a good theory, we should just change that to use the SafeDelete

Cheers,
Rico

**Comment by sigmundch** [comment]
never mind - that code is unreachable :(

**Comment by ricowind** [comment]
I will then change my response:
We should change that code to the empty string

**Comment by sigmundch** [comment]
+1 :) 

OK, we logged into the bots and now we know what the issue is: the path names were too long! Noooooo!!! 

So we probably need similar fixes like @whesse  did in the normal Dart repo.

**Comment by jakemac53** [comment]
http://bit.ly/1HVJkxS
